### PR TITLE
Config parser

### PR DESCRIPTION
**Pull request by dwfncar**
- Read CAMP files list from a top-level config file named config.json (should this be made a parameter to _JsonReaderPolicy::Parse_, and not hard-coded?).
- If config.json does not exist in the config directory, default config files will be used, as previously defined.
- Added additional tests in test/unit/test_solver_config.cpp.
- **Note**: for now am assuming the _camp-list_ has only two entries and is ordered as a species file followed by a mechanism/reactions file.
- To relax this assumption we could open each file in the CAMP list and search for JSON keys that determine the type of config file.  But based on previous discussions we will likely define a **new task** to re-organize the JSON parser, simply merging all config files in the CAMP list and simultaneously parsing species and reactions https://github.com/NCAR/micm/issues/256 .

### CODE REVIEW

File: include/micm/configure/solver_config.hpp [hunk 1]
```diff
@@ -122,9 +128,35 @@ namespace micm
...
       std::filesystem::path reactions_config(config_dir / REACTIONS_CONFIG);
+      // Note tolerance_config is defined here but not used
       std::filesystem::path tolerance_config(config_dir / TOLERANCE_CONFIG);
 
-      // Current reaction configs should be either mechanism_config or reactions config
+      // Look for CAMP config file
+      std::filesystem::path camp_config(config_dir / CAMP_CONFIG);
+      if (std::filesystem::exists(camp_config))
+      {
+        std::cout << "Reading CAMP config " << camp_config << std::endl;
```
Comment (K20shores) [inline include/micm/configure/solver_config.hpp:138]: Please remove or comment this out. Hopefully we will be adding a logger when we do #127, but for now I'd prefer we keep output relatively clear.

File: include/micm/configure/solver_config.hpp [hunk 2]
```diff
@@ -122,9 +128,35 @@ namespace micm
...
+        json camp_data = json::parse(std::ifstream(camp_config));
+        if (!camp_data.contains(CAMP_FILES))
+          return ConfigParseStatus::CAMPFilesSectionNotFound;
+
+        std::vector<std::string> camp_files;
+        for (const auto& element : camp_data[CAMP_FILES])
+        {
+          std::cout << element.get<std::string>() << std::endl;
```
Comment (K20shores) [inline include/micm/configure/solver_config.hpp:146]: same as above

**Review by K20shores — CHANGES_REQUESTED** [review]

**Review by boulderdaze — APPROVED** [review]

File: include/micm/configure/solver_config.hpp [hunk 3]
```diff
@@ -122,9 +128,35 @@ namespace micm
...
       std::filesystem::path reactions_config(config_dir / REACTIONS_CONFIG);
+      // Note tolerance_config is defined here but not used
       std::filesystem::path tolerance_config(config_dir / TOLERANCE_CONFIG);
 
-      // Current reaction configs should be either mechanism_config or reactions config
+      // Look for CAMP config file
+      std::filesystem::path camp_config(config_dir / CAMP_CONFIG);
+      if (std::filesystem::exists(camp_config))
+      {
+        // std::cout << "Reading CAMP config " << camp_config << std::endl;
```
Comment (mattldawson) [inline include/micm/configure/solver_config.hpp:138]: can remove commented-out line

File: include/micm/configure/solver_config.hpp [hunk 4]
```diff
@@ -122,9 +128,35 @@ namespace micm
...
+        json camp_data = json::parse(std::ifstream(camp_config));
+        if (!camp_data.contains(CAMP_FILES))
+          return ConfigParseStatus::CAMPFilesSectionNotFound;
+
+        std::vector<std::string> camp_files;
+        for (const auto& element : camp_data[CAMP_FILES])
+        {
+          // std::cout << element.get<std::string>() << std::endl;
```
Comment (mattldawson) [inline include/micm/configure/solver_config.hpp:146]: can remove commented-out line

**Review by mattldawson — APPROVED** [review]
looks great! just two very minor comments

**Review by K20shores — APPROVED** [review]

**Pull request merged**